### PR TITLE
[Lab 1] Create Python and Swift web applications

### PR DESCRIPTION
**Pull request by igooor-bb**
In this PR I created web applications in Python and Swift and added accompanying markdown files.

### CODE REVIEW

File: app_python/PYTHON.md [hunk 1]
```diff
@@ -0,0 +1,23 @@
+# About best practices
+
+## Web application in Python
+
+When choosing a framework for a web application I had to choose between [Flask](https://flask.palletsprojects.com/en/2.2.x/) and [Django](https://www.djangoproject.com), and in the end I chose the first one because it is lightweight and more suitable for small applications.
+
```
Comment (ghost) [inline app_python/PYTHON.md:8]: Where is pros and cons of framework?

File: app_swift/SWIFT.md [hunk 2]
```diff
@@ -0,0 +1,21 @@
+# About best practices
+
+## Web application in Swift
+
+It did not take me long to choose a framework for a web application, because the strongest and most stable Swift framework is [Vapor](https://vapor.codes).
```
Comment (ghost) [inline app_swift/SWIFT.md:7]: where are pros and cons of framework?

File: app_python/PYTHON.md [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+- Use a virtual environment and `requirements.txt` that allows you to automatically pull up the dependencies you use when you deploy.
+
+- Use `.gitignore` to avoid bringing extra files into the repository (such as a local virtual environment).
+
+- Use [Docker](https://www.docker.com) to have and provide the ability to quickly run the service on any computer.
+
+- Use `.dockerignore` to avoid bringing extra files into the container (such as markdown files and virtual environment).
+
+- Use [uWSGI](https://uwsgi-docs.readthedocs.io/en/latest/) to run Flask application and [nginx](https://www.nginx.com) as a reverse proxy to it.
```
Comment (markovvn1) [inline app_python/PYTHON.md:35]: You also use Jinja for templates, which is good practice

File: app_swift/Dockerfile [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+WORKDIR /app
+
+COPY --from=build --chown=vapor:vapor /staging /app
+
+USER vapor:vapor
+
+EXPOSE 8080
+
+ENTRYPOINT ["./Run"]
+CMD ["serve", "--env", "production", "--hostname", "0.0.0.0", "--port", "8080"]
```
Comment (markovvn1) [inline app_swift/Dockerfile:47]: The application seems to be looking for `/flutter_service_worker.js` file but can't find it. Is it ok?
image

Comment (igooor-bb) [reply app_swift/Dockerfile:47]: I have no idea why Swift-based framework is complaining about some weird Flutter file 😳

File: app_python/run.sh [hunk 5]
```diff
@@ -0,0 +1,3 @@
+#!/usr/bin/env bash
+service nginx start
+uwsgi --ini uwsgi.ini
```
Comment (AzazKamaz) [inline app_python/run.sh:3]: It is best practice to run only one process inside docker container

**Review by AzazKamaz — APPROVED** [review]

File: app_swift/Dockerfile (hunk 4 above)
Comment (markovvn1) [reply app_swift/Dockerfile:47]: You're right. It seems that before your review, I reviewed the software on flatter and the browser remembered this path and accessed it by itself. I ran the program again, now there is no problem.

image

File: app_python/run.sh (hunk 5 above)
Comment (markovvn1) [reply app_python/run.sh:3]: Best practices of Docker is the next lab)

**Comment by glebosotov** [comment]
LGTM :)
👍🏻 for Swift

**Review by ghost — APPROVED** [review]

**Pull request closed without merging**